src/crypto.rs

```diff
@@ -165,8 +157,14 @@ where
     }
 }
 
-fn derive_key(salt: &Salt, passphrase: &SecUtf8, params: &scrypt::ScryptParams) -> [u8; 32] {
+fn derive_key(salt: &Salt, passphrase: &SecUtf8) -> [u8; 32] {
     let mut key = [0u8; 32];
+
+    #[cfg(any(test, feature = "test"))]
+    let params = *KDF_PARAMS_TEST;
+    #[cfg(not(any(test, feature = "test")))]
+    let params = *KDF_PARAMS_PROD;
```

Comment:
Would it make sense to have only one static `KDF_PARAMS`, but initialized with different values based on the feature?

Reply:
Think so too!

Reply:
Not so sure this will improve much, besides pleasing the disciples of the DRY goods. If we leave it like this the signal it carries for something like an auditor is invaluable - a dimension we should optimise for.

Reply:
Not only DRY orthodox, but also making it statically impossible to use the wrong value. In this setup even when compiling for production, there's an insecure test configuration value lurking somewhere deep in the code just waiting to step in and ruin the day.

Reply:
That's a great point. Yet for someone to understand which values are used depends on their knowledge of how these flags propagate, something we as a group already struggled to tame and make usable.

Reply:
We're already using this mechanism, I'm only proposing to move its usage

Reply:
Interestingly, none of you brought up that this change prevents migrating to different parameters in the future. If they change, existing keystores will be rendered unusable. Having the `Pwhash` parametrised over them would allow users of the library to migrate them. :shrug:

Reply:
But this doesn't stop us from passing `Pwhash` parameters around. We can make value of `KDF_PARAMS` dependent on the cfg flag and stop choosing between `KDF_PARAMS_PROD` and `KDF_PARAMS_TEST` during runtime while still explicitly passing its value everywhere.